packages/api/src/objectSet/ObjectSet.test.ts

```diff
@@ -52,13 +80,127 @@ describe("ObjectSet", () => {
+      const fetchPageResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({
+          // @ts-expect-error
+          $includeAllBaseObjectProperties: true,
+          $select: ["attachment"],
+        });
+
+      const asyncIterResult = fauxObjectSet
+        .where({ class: "idk" })
+        .asyncIter({
+          // @ts-expect-error
+          $includeAllBaseObjectProperties: true,
+          $select: ["attachment"],
+        });
+    });
+
+    it("does let you pass full select options and false", async () => {
+      const fetchPageResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({
+          $includeAllBaseObjectProperties: true,
+
+          // this select list is intended to represent all properties on `EmployeeApiTest`,
+          // so if you get an error here later and you added properties to that object,
+          // be sure to add them here too.
+          $select: [
+            "attachment",
+            "class",
+            "employeeId",
+            "fullName",
+            "geopoint",
+            "geotimeSeriesReference",
+            "isActive",
+            "mediaReference",
+            "timeseries",
+          ],
+        });
+
+      const asyncIterResult = await fauxObjectSet
+        .where({ class: "idk" })
+        .fetchPage({
+          $includeAllBaseObjectProperties: true,
+
+          // this select list is intended to represent all properties on `EmployeeApiTest`,
+          // so if you get an error here later and you added properties to that object,
+          // be sure to add them here too.
+          $select: [
+            "attachment",
+            "class",
+            "employeeId",
+            "fullName",
+            "geopoint",
+            "geotimeSeriesReference",
+            "isActive",
+            "mediaReference",
+            "timeseries",
+          ],
+        });
+    });
+  });
+
+  test("includeRid", async () => {
+    const x = await fauxObjectSet
+      .where({ class: "idk" })
+      .fetchPage({ $includeRid: true });
+
+    expectTypeOf(x).toEqualTypeOf<
+      PageResult<
+        Osdk.Instance<EmployeeApiTest, "$rid">
+      >
+    >();
+  });
+
+  test("pivotTo", async () => {
+    const noArgs = await fauxObjectSet.pivotTo("peeps").fetchPage({});
+    const subselect = await fauxObjectSet.pivotTo("peeps").fetchPage({
+      $select: ["employeeId", "class"],
+    });
+
+    expectTypeOf(subselect).toEqualTypeOf<
+      PageResult<
+        Osdk.Instance<EmployeeApiTest, never, "employeeId" | "class">
+      >
+    >();
+  });
+
   describe(".withProperties", () => {
     test("single property", async () => {
       const withA = fauxObjectSet.withProperties({
         "a": (base) => {
           return base.pivotTo("lead").aggregate("class:exactDistinct");
         },
       });
+      withA.pivotTo("lead");
+
+      const isWithAAssignable: $ObjectSet<EmployeeApiTest, {}> = withA;
```

Comment:
One of the weird things we saw from the bug report was the way the inference happened if it was assigned right away from a return result or if it was assigned after the fact. Just bolstering things here.